### PR TITLE
Vigenere Cipher Explanation added in English language

### PR DESCRIPTION
**Pull request by JahnabDutta**
### Description

Provided the explanation for vignere cipher in the cipher directory of the English language. Added brief encryption and decryption algorithm along with an example. Then described the complexity analysis of the scheme. Furthermore, added resources about regarding the cryptanalysis(attack) of the cipher and caution regarding not using it for real life purposes. 
Also attached the implementation links in python, C++ and javascript. 

### Checklist

- [x] A description of the changes in this PR is mentioned above.
- [x] All the new and existing checks pass.
- [x] No plagiarized, duplicated, or repetitive documentation that has been directly copied from another source.
- [x] If it's a new explanation, it contains solid, understandable, and accessible information.
- [x] I have read the whole [contributing guidelines](https://github.com/TheAlgorithms/Algorithms-Explanation/blob/master/CONTRIBUTING.md) and agree to the [Code of Conduct](https://github.com/TheAlgorithms/.github/blob/master/CODE_OF_CONDUCT.md).

### CODE REVIEW

File: en/Ciphers/vignere_cipher.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+1. Time Complexity: `O(n)`
+2. Space Complexity: `O(n)`
+
+## Cryptanalysis and caution
+
+1. It is a toy cipher. It is easy to crack. It is not secure.
+2. There are several key finding methods such as
+    - [Kasiski examination](https://en.wikipedia.org/wiki/Kasiski_examination)
+    - [Index of coincidence](https://en.wikipedia.org/wiki/Index_of_coincidence)
+3. Once key is found, we can use [frequency analysis](https://en.wikipedia.org/wiki/Frequency_analysis) to crack the cipher.
```
Comment (appgurueu) [inline en/Ciphers/vignere_cipher.md:72]: Once the *key length* is found.

**Review by appgurueu — COMMENTED** [review]
Otherwise seems to be fine content-wise. Should perhaps point out the relation to the Caeser Cipher?

**Comment by JahnabDutta** [comment]
Made changes and also added fixed spelling errors.

**Review by appgurueu — APPROVED** [review]
I've now done a pass to fix typos and improve the formatting and wording a bit.

File: en/Ciphers/vigenere_cipher.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
+# Vigenere Cipher
+
+The Vigenere cipher is a famous toy cipher. It was invented in 1553 by the Italian cryptographer Giovan Battista Bellaso but for centuries was attributed to the 16th-century French cryptographer Blaise de Vigenère, who devised a similar cipher in 1586.
+
+It is easy to encrypt and decrypt but at the same time, it is easy to intercept and crack this cipher. It is a polyalphabetic substitution cipher, meaning that the same letter of the alphabet can be replaced by  different letters depending on the key. The key is a word or a phrase that is repeated to match the length of the message. Then the letters of the key are used to shift the letters of the message.
```
Comment (Panquesito7) [inline en/Ciphers/vigenere_cipher.md:5]:
```suggestion
It is easy to encrypt and decrypt but at the same time, it is easy to intercept and crack this cipher. It is a polyalphabetic substitution cipher, meaning that the same letter of the alphabet can be replaced by different letters depending on the key. The key is a word or phrase that is repeated to match the length of the message. Then the letters of the key are used to shift the letters of the message.
```

File: en/Ciphers/vigenere_cipher.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+## Improvement over Caesar cipher
+
+The Vigenere cipher is an improvement over the Caesar cipher. The Caesar cipher uses a single character as the key to shift the message. In contrast, the Vigenere cipher uses a word or a phrase as the key. This means that the same character occurring at different positions in the message will be shifted by different amounts. This makes it harder to crack the cipher.
```
Comment (Panquesito7) [inline en/Ciphers/vigenere_cipher.md:9]:
```suggestion
The Vigenere cipher is an improvement over the Caesar cipher. The Caesar cipher uses a single character as the key to shifting the message. In contrast, the Vigenere cipher uses a word or a phrase as the key. This means that the same character occurring at different positions in the message will be shifted by different amounts. This makes it harder to crack the cipher.
```

File: en/Ciphers/vigenere_cipher.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+2. There are several key (length) finding methods such as
+    - [Kasiski examination](https://en.wikipedia.org/wiki/Kasiski_examination)
+    - [Index of coincidence](https://en.wikipedia.org/wiki/Index_of_coincidence)
+3. Once the key length is found, [frequency analysis](https://en.wikipedia.org/wiki/Frequency_analysis) can be used to find the key and hence crack the cipher.
+4. Therefore, this cipher should not be used to encrypt any important data.
+
+## Implementations
+
+- [C++](https://github.com/TheAlgorithms/Python/blob/master/ciphers/vigenere_cipher.py)
+- [Python](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/ciphers/vigenere_cipher.cpp)
```
Comment (Panquesito7) [inline en/Ciphers/vigenere_cipher.md:68]: Links seem to be inverted.
```suggestion
- [C++](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/ciphers/vigenere_cipher.cpp)
- [Python](https://github.com/TheAlgorithms/Python/blob/master/ciphers/vigenere_cipher.py)
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

File: en/Ciphers/vigenere_cipher.md (hunk 3 above)
Comment (appgurueu) [reply en/Ciphers/vigenere_cipher.md:9]: I think "shift" is fine here.

**Review by appgurueu — APPROVED** [review]

**Review by Panquesito7 — APPROVED** [review]
Thanks! 🚀

**Pull request merged**